### ISSUE TITLE
Tests failing locally

### ISSUE THREAD

**Issue by AlanRosenthal** [google/bumble#81]
on commit `ae0b739e4a7784e5f8f5592597e51330c03d0de0`
```
(Bumble) ➜  bumble git:(main) invoke test          
============================= test session starts ==============================
platform linux -- Python 3.8.15, pytest-7.2.0, pluggy-1.0.0
rootdir: /home/alanrosenthal/code/fitbit/bumble/tests, configfile: pytest.ini
plugins: metadata-2.0.4, asyncio-0.20.2, html-3.2.0
asyncio: mode=auto
collected 395 items

tests/a2dp_test.py ..                                                    [  0%]
tests/avdtp_test.py ..                                                   [  1%]
tests/core_test.py .                                                     [  1%]
tests/device_test.py F                                                   [  1%]
tests/gatt_test.py ..........                                            [  4%]
tests/hci_test.py ................................                       [ 12%]
tests/import_test.py .F.                                                 [ 12%]
tests/l2cap_test.py ....                                                 [ 13%]
tests/rfcomm_test.py .                                                   [ 14%]
tests/sdp_test.py .                                                      [ 14%]
tests/self_test.py ..................................................... [ 27%]
........................................................................ [ 46%]
........................................................................ [ 64%]
........................................................................ [ 82%]
........................................................                 [ 96%]
tests/smp_test.py ...........                                            [ 99%]
tests/transport_test.py ..                                               [100%]

=================================== FAILURES ===================================
_________________________ test_device_connect_parallel _________________________

    @pytest.mark.asyncio
    async def test_device_connect_parallel():
        d0 = Device(host=Host(None, None))
        d1 = Device(host=Host(None, None))
        d2 = Device(host=Host(None, None))
    
        # enable classic
        d0.classic_enabled = True
        d1.classic_enabled = True
        d2.classic_enabled = True
    
        # set public addresses
        d0.public_address = Address('F0:F1:F2:F3:F4:F5', address_type=Address.PUBLIC_DEVICE_ADDRESS)
        d1.public_address = Address('F5:F4:F3:F2:F1:F0', address_type=Address.PUBLIC_DEVICE_ADDRESS)
        d2.public_address = Address('F5:F4:F3:F3:F4:F5', address_type=Address.PUBLIC_DEVICE_ADDRESS)
    
        def d0_flow():
            packet = HCI_Packet.from_bytes((yield))
            assert packet.name == 'HCI_CREATE_CONNECTION_COMMAND'
            assert packet.bd_addr == d1.public_address
    
            d0.host.on_hci_packet(HCI_Command_Status_Event(
                status                  = HCI_COMMAND_STATUS_PENDING,
                num_hci_command_packets = 1,
                command_opcode          = HCI_CREATE_CONNECTION_COMMAND
            ))
    
            d1.host.on_hci_packet(HCI_Connection_Request_Event(
               bd_addr         = d0.public_address,
               class_of_device = 0,
               link_type       = HCI_Connection_Complete_Event.ACL_LINK_TYPE
            ))
    
            packet = HCI_Packet.from_bytes((yield))
            assert packet.name == 'HCI_CREATE_CONNECTION_COMMAND'
            assert packet.bd_addr == d2.public_address
    
            d0.host.on_hci_packet(HCI_Command_Status_Event(
                status                  = HCI_COMMAND_STATUS_PENDING,
                num_hci_command_packets = 1,
                command_opcode          = HCI_CREATE_CONNECTION_COMMAND
            ))
    
            d2.host.on_hci_packet(HCI_Connection_Request_Event(
               bd_addr         = d0.public_address,
               class_of_device = 0,
               link_type       = HCI_Connection_Complete_Event.ACL_LINK_TYPE
            ))
    
            assert (yield) == None
    
        def d1_flow():
            packet = HCI_Packet.from_bytes((yield))
            assert packet.name == 'HCI_ACCEPT_CONNECTION_REQUEST_COMMAND'
    
            d1.host.on_hci_packet(HCI_Command_Complete_Event(
                num_hci_command_packets = 1,
                command_opcode          = HCI_ACCEPT_CONNECTION_REQUEST_COMMAND,
                return_parameters       = b"\x00"
            ))
    
            d1.host.on_hci_packet(HCI_Connection_Complete_Event(
                status             = HCI_SUCCESS,
                connection_handle  = 0x100,
                bd_addr            = d0.public_address,
                link_type          = HCI_Connection_Complete_Event.ACL_LINK_TYPE,
                encryption_enabled = True,
            ))
    
            d0.host.on_hci_packet(HCI_Connection_Complete_Event(
                status             = HCI_SUCCESS,
                connection_handle  = 0x100,
                bd_addr            = d1.public_address,
                link_type          = HCI_Connection_Complete_Event.ACL_LINK_TYPE,
                encryption_enabled = True,
            ))
    
            assert (yield) == None
    
        def d2_flow():
            packet = HCI_Packet.from_bytes((yield))
            assert packet.name == 'HCI_ACCEPT_CONNECTION_REQUEST_COMMAND'
    
            d2.host.on_hci_packet(HCI_Command_Complete_Event(
                num_hci_command_packets = 1,
                command_opcode          = HCI_ACCEPT_CONNECTION_REQUEST_COMMAND,
                return_parameters       = b"\x00"
            ))
    
            d2.host.on_hci_packet(HCI_Connection_Complete_Event(
                status             = HCI_SUCCESS,
                connection_handle  = 0x101,
                bd_addr            = d0.public_address,
                link_type          = HCI_Connection_Complete_Event.ACL_LINK_TYPE,
                encryption_enabled = True,
            ))
    
            d0.host.on_hci_packet(HCI_Connection_Complete_Event(
                status             = HCI_SUCCESS,
                connection_handle  = 0x101,
                bd_addr            = d2.public_address,
                link_type          = HCI_Connection_Complete_Event.ACL_LINK_TYPE,
                encryption_enabled = True,
            ))
    
            assert (yield) == None
    
        d0.host.set_packet_sink(Sink(d0_flow()))
        d1.host.set_packet_sink(Sink(d1_flow()))
        d2.host.set_packet_sink(Sink(d2_flow()))
    
>       [c01, c02, a10, a20, a01] = await asyncio.gather(*[
            asyncio.create_task(d0.connect(d1.public_address, transport=BT_BR_EDR_TRANSPORT)),
            asyncio.create_task(d0.connect(d2.public_address, transport=BT_BR_EDR_TRANSPORT)),
            asyncio.create_task(d1.accept(peer_address=d0.public_address)),
            asyncio.create_task(d2.accept()),
            asyncio.create_task(d0.accept(peer_address=d1.public_address)),
        ])

tests/device_test.py:161: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
bumble/device.py:1387: in accept
    if peer_address == Address.NIL:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <bumble.hci.Address object at 0x7f4fa68f2e80>
other = <bound method ? of <class 'bumble.hci.Address'>>

    def __eq__(self, other):
>       return self.address_bytes == other.address_bytes and self.is_public == other.is_public
E       AttributeError: 'property' object has no attribute 'address_bytes'

bumble/hci.py:1750: AttributeError
_______________________________ test_app_imports _______________________________

    def test_app_imports():
>       from bumble.apps.console import main
E       ModuleNotFoundError: No module named 'bumble.apps'

tests/import_test.py:67: ModuleNotFoundError
=============================== warnings summary ===============================
self_test.py::test_self_smp_reject
self_test.py::test_self_smp_reject
self_test.py::test_self_smp_wrong_pin
self_test.py::test_self_smp_wrong_pin
self_test.py::test_self_smp_wrong_pin
self_test.py::test_self_smp_wrong_pin
  /home/alanrosenthal/code/fitbit/bumble/bumble/smp.py:1113: DeprecationWarning: The 'warn' method is deprecated, use 'warning' instead
    logger.warn(f'pairing failure ({error_name(reason)})')

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=========================== short test summary info ============================
FAILED tests/device_test.py::test_device_connect_parallel - AttributeError: '...
FAILED tests/import_test.py::test_app_imports - ModuleNotFoundError: No modul...
================== 2 failed, 393 passed, 6 warnings in 22.39s ==================
```

**Comment by AlanRosenthal** [comment]
CI runs py3.10, env.yml is set to 3.8, perhaps that's related?

**Comment by AlanRosenthal** [comment]
confirmed py3.8 fails, 3.9+ passes: https://github.com/google/bumble/pull/83

**Comment by uael** [comment]
Good catch, I guess the magic of combining `@classmethod` & `@property` come to an end 🥲

I used that trick without paying attention to portability, it's in fact not supported prior 3.9 and is then [removed in 3.11](
https://docs.python.org/3.11/library/functions.html?highlight=classmethod#classmethod).

I'm definitely gonna fix that next week!

**Comment by uael** [comment]
Fixed in #89
